--- ./usr/include/stdlib.h.orig	2021-08-16 11:50:47.432239182 +0200
+++ ./usr/include/stdlib.h	2021-08-16 11:51:31.102222894 +0200
@@ -33,7 +33,11 @@
 #include <bits/wait.h>
 #include <malloc.h>
 #include <stddef.h>
+#if !defined(__swift__)
+#include <stdint.h>
+#endif
 #include <sys/cdefs.h>
+#include <sys/sysinfo.h>
 #include <xlocale.h>
 
 __BEGIN_DECLS
@@ -191,9 +195,18 @@
  * Returns the number of samples written to `__averages` (at most 3), and returns -1 on failure.
  */
 
-#if __ANDROID_API__ >= 29
-int getloadavg(double __averages[_Nonnull], int __n) __INTRODUCED_IN(29);
-#endif /* __ANDROID_API__ >= 29 */
+static __inline__ int getloadavg(double __averages[_Nonnull], int __n) {
+  if (__n < 0) return -1;
+  if (__n > 3) __n = 3;
+
+  struct sysinfo si;
+  if (sysinfo(&si) == -1) return -1;
+
+  for (int i = 0; i < __n; ++i) {
+    __averages[i] = (double)si.loads[i] / (double)(1 << SI_LOAD_SHIFT);
+  }
+  return __n;
+}
 
 
 /* BSD compatibility. */
@@ -207,8 +220,7 @@
 
 size_t wcstombs(char* _Nullable __dst, const wchar_t* _Nullable __src, size_t __n);
 
-size_t __ctype_get_mb_cur_max(void);
-#define MB_CUR_MAX __ctype_get_mb_cur_max()
+#define MB_CUR_MAX 4
 
 #if defined(__BIONIC_INCLUDE_FORTIFY_HEADERS)
 #include <bits/fortify/stdlib.h>
