--- a/gcc/configure.orig	2024-07-10 06:08:14.093525767 -0400
+++ b/gcc/configure	2024-07-10 06:09:39.136867572 -0400
@@ -29547,7 +29547,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $gcc_cv_c_no_fpie" >&5
 $as_echo "$gcc_cv_c_no_fpie" >&6; }
 if test "$gcc_cv_c_no_fpie" = "yes"; then
-  NO_PIE_CFLAGS="-fno-PIE"
+  NO_PIE_CFLAGS=""
 fi
 
 
@@ -29575,7 +29575,7 @@
 { $as_echo "$as_me:${as_lineno-$LINENO}: result: $gcc_cv_no_pie" >&5
 $as_echo "$gcc_cv_no_pie" >&6; }
 if test "$gcc_cv_no_pie" = "yes"; then
-  NO_PIE_FLAG="-no-pie"
+  NO_PIE_FLAG=""
 fi
 
 
