--- a/lib/jxl/base/data_parallel.orig.h	2024-09-19 15:10:46.456259549 -0400
+++ b/lib/jxl/base/data_parallel.h	2024-09-19 15:08:16.912418083 -0400
@@ -10,6 +10,8 @@
 // data-parallel computations.
 
 #include <jxl/parallel_runner.h>
+/*#include "lib/include/jxl/parallel_runner.h"*/
+#define JXL_PARALLEL_RET_SUCCESS (0)
 
 #include <atomic>
 #include <cstddef>
--- a/lib/include/jxl/parallel_runner.h
+++ b/lib/include/jxl/parallel_runner.h
@@ -54,7 +54,7 @@
 /**
  * Code returned by the @ref JxlParallelRunInit function to indicate success.
  */
-#define JXL_PARALLEL_RET_SUCCESS (0)
+#define JXL_PARALLEL_RET_SUCCESS 0
 
 /**
  * Code returned by the @ref JxlParallelRunInit function to indicate a general
