--- a/lib/include/jxl/decode.h
+++ a/lib/include/jxl/decode.h
@@ -39,7 +39,7 @@
 
 /** The result of @ref JxlSignatureCheck.
  */
-typedef enum {
+typedef enum JxlSignature {
   /** Not enough bytes were passed to determine if a valid signature was found.
    */
   JXL_SIG_NOT_ENOUGH_BYTES = 0,
@@ -55,7 +55,7 @@
   /** A valid container signature was found, that is a JPEG XL image embedded
    * in a box format container.
    */
-  JXL_SIG_CONTAINER = 3,
+  JXL_SIG_CONTAINER = 3
 } JxlSignature;
 
 /**
@@ -119,7 +119,7 @@
  * events that can be subscribed to, they are never returned if they
  * have not been registered with @ref JxlDecoderSubscribeEvents.
  */
-typedef enum {
+typedef enum JxlDecoderStatus {
   /** Function call finished successfully, or decoding is finished and there is
    * nothing more to be done.
    *
@@ -323,7 +323,7 @@
   /** The box being decoded is now complete. This is only emitted if a buffer
    * was set for the box.
    */
-  JXL_DEC_BOX_COMPLETE = 0x10000,
+  JXL_DEC_BOX_COMPLETE = 0x10000
 } JxlDecoderStatus;
 
 /** Types of progressive detail.
@@ -334,7 +334,7 @@
  *  - @ref kLastPasses (which implies @ref kDC and @ref kFrames)
  *  - @ref kPasses (which implies @ref kLastPasses, kDC and @ref kFrames)
  */
-typedef enum {
+typedef enum JxlProgressiveDetail {
   /**
    * after completed kRegularFrames
    */
@@ -364,7 +364,7 @@
   /**
    * after completed groups
    */
-  kGroups = 6,
+  kGroups = 6
 } JxlProgressiveDetail;
 
 /** Rewinds decoder to the beginning. The same input must be given again from
@@ -710,13 +710,13 @@
  * or the color profile from the pixel data produced by the decoder. Both are
  * the same if the JxlBasicInfo has uses_original_profile set.
  */
-typedef enum {
+typedef enum JxlColorProfileTarget {
   /** Get the color profile of the original image from the metadata.
    */
   JXL_COLOR_PROFILE_TARGET_ORIGINAL = 0,
 
   /** Get the color profile of the pixel data the decoder outputs. */
-  JXL_COLOR_PROFILE_TARGET_DATA = 1,
+  JXL_COLOR_PROFILE_TARGET_DATA = 1
 } JxlColorProfileTarget;
 
 /**
