--- a/libatomic/libatomic_i.h
+++ b/libatomic/libatomic_i.h
@@ -127,11 +127,15 @@
 
 /* Utility macros for GCC attributes.  */
 #define UNUSED          __attribute__((unused))
+#if defined(__GNUC__)
+# define HIDDEN
+#else
 #ifdef HAVE_ATTRIBUTE_VISIBILITY
 # define HIDDEN         __attribute__((visibility("hidden")))
 #else
 # define HIDDEN
 #endif
+#endif
 
 /* Occasionally we have to play games with internal and external symbol
    names, in order to work around builtin functions of the same name.
