--- gcc-6.1.0/libcpp/system.h	2016-01-04 09:30:50.000000000 -0500
+++ gcc-6.1.0_my/libcpp/system.h	2024-05-13 22:19:06.612131469 -0400
@@ -375,9 +375,12 @@
    ??? C99 designated initializers are not supported by most C++
    compilers, including G++.  -- gdr, 2005-05-18  */
 #if !defined(HAVE_DESIGNATED_INITIALIZERS)
+#if !defined(__cplusplus)
 #define HAVE_DESIGNATED_INITIALIZERS \
-  (!defined(__cplusplus) \
-   && ((GCC_VERSION >= 2007) || (__STDC_VERSION__ >= 199901L)))
+   ((GCC_VERSION >= 2007) || (__STDC_VERSION__ >= 199901L))
+#else
+#define HAVE_DESIGNATED_INITIALIZERS 0
+#endif
 #endif
 
 #ifndef offsetof
