diff -Naur gcc-4.9.2/gcc/system.h gcc-4.9.2_termux/gcc/system.h
--- a/gcc/system.h	2014-01-02 17:23:26.000000000 -0500
+++ b/gcc/system.h	2024-02-21 23:47:37.877323689 -0500
@@ -259,6 +259,7 @@
    an error about 'using std::malloc'.  */
 #ifdef __cplusplus
 #include <cstdlib>
+#include <map>
 #endif
 
 /* Undef vec_free from AIX stdlib.h header which conflicts with vec.h.  */
@@ -570,15 +571,21 @@
 
 /* 1 if we have C99 designated initializers.  */
 #if !defined(HAVE_DESIGNATED_INITIALIZERS)
+# if !defined(__cplusplus)
-#define HAVE_DESIGNATED_INITIALIZERS \
-  (((GCC_VERSION >= 2007) || (__STDC_VERSION__ >= 199901L)) \
-   && !defined(__cplusplus ))
+#  define HAVE_DESIGNATED_INITIALIZERS \
+    ((GCC_VERSION >= 2007) || (__STDC_VERSION__ >= 199901L))
+# else
+#  define HAVE_DESIGNATED_INITIALIZERS false
+# endif
 #endif
 
 #if !defined(HAVE_DESIGNATED_UNION_INITIALIZERS)
+# if !defined(__cplusplus) || (GCC_VERSION >= 4007)
-#define HAVE_DESIGNATED_UNION_INITIALIZERS \
-  (((GCC_VERSION >= 2007) || (__STDC_VERSION__ >= 199901L)) \
-   && (!defined(__cplusplus) || (GCC_VERSION >= 4007)))
+#  define HAVE_DESIGNATED_UNION_INITIALIZERS \
+    ((GCC_VERSION >= 2007) || (__STDC_VERSION__ >= 199901L))
+# else
+#  define HAVE_DESIGNATED_UNION_INITIALIZERS false
+# endif
 #endif
 
 #if HAVE_SYS_STAT_H
