diff -Naur gcc-5.3.0/include/libiberty.h gcc-5.3.0_my/include/libiberty.h
--- gcc-5.3.0/include/libiberty.h	2015-01-14 16:55:35.000000000 -0500
+++ gcc-5.3.0_my/include/libiberty.h	2024-06-10 05:50:38.383568029 -0400
@@ -104,7 +104,7 @@
    declaration without arguments.  If it is 0, we checked and failed
    to find the declaration so provide a fully prototyped one.  If it
    is 1, we found it so don't provide any declaration at all.  */
-#if !HAVE_DECL_BASENAME
+#if defined HAVE_DECL_BASENAME && !HAVE_DECL_BASENAME
 #if defined (__GNU_LIBRARY__ ) || defined (__linux__) \
  || defined (__FreeBSD__) || defined (__OpenBSD__) || defined (__NetBSD__) \
  || defined (__CYGWIN__) || defined (__CYGWIN32__) || defined (__MINGW32__) \
