diff -Naur gcc-5.3.0/fixincludes/system.h gcc-5.3.0_my/fixincludes/system.h
--- gcc-5.3.0/fixincludes/system.h	2009-04-09 11:00:19.000000000 -0400
+++ gcc-5.3.0_my/fixincludes/system.h	2024-05-21 23:10:38.116038535 -0400
@@ -73,6 +73,7 @@
 #  undef getchar
 #  define getchar() getchar_unlocked ()
 # endif
+#undef HAVE_FPUTC_UNLOCKED
 # ifdef HAVE_FPUTC_UNLOCKED
 #  undef fputc
 #  define fputc(C, Stream) fputc_unlocked (C, Stream)
diff -Naur gcc-5.3.0/libcpp/system.h gcc-5.3.0_my/libcpp/system.h
--- gcc-5.3.0/libcpp/system.h	2015-01-05 07:33:28.000000000 -0500
+++ gcc-5.3.0_my/libcpp/system.h	2024-05-21 23:11:34.136044115 -0400
@@ -86,6 +86,7 @@
 #  undef getchar
 #  define getchar() getchar_unlocked ()
 # endif
+#undef HAVE_FPUTC_UNLOCKED
 # ifdef HAVE_FPUTC_UNLOCKED
 #  undef fputc
 #  define fputc(C, Stream) fputc_unlocked (C, Stream)
