diff -Naur gcc-4.9.2/gcc/bitmap.c gcc-4.9.2_curr/gcc/bitmap.c
--- ../gcc-4.9.2/gcc/bitmap.c	2014-01-02 17:23:26.000000000 -0500
+++ ./gcc/bitmap.c	2024-03-04 05:45:22.294576439 -0500
@@ -2166,9 +2166,9 @@
       s[41] = 0;
       fprintf (stderr,
 	       "%-41s %9u"
-	       " %15"HOST_WIDEST_INT_PRINT"d %15"HOST_WIDEST_INT_PRINT"d"
-	       " %15"HOST_WIDEST_INT_PRINT"d"
-	       " %10"HOST_WIDEST_INT_PRINT"d %10"HOST_WIDEST_INT_PRINT"d\n",
+	       " %15" HOST_WIDEST_INT_PRINT "d %15" HOST_WIDEST_INT_PRINT "d"
+	       " %15" HOST_WIDEST_INT_PRINT "d"
+	       " %10" HOST_WIDEST_INT_PRINT "d %10" HOST_WIDEST_INT_PRINT "d\n",
 	       s, d->created,
 	       d->allocated, d->peak, d->current,
 	       d->nsearches, d->search_iter);
@@ -2201,7 +2201,7 @@
   bitmap_desc_hash.traverse <output_info *, print_statistics> (&info);
   fprintf (stderr, "---------------------------------------------------------------------------------\n");
   fprintf (stderr,
-	   "%-41s %9"HOST_WIDEST_INT_PRINT"d %15"HOST_WIDEST_INT_PRINT"d\n",
+	   "%-41s %9" HOST_WIDEST_INT_PRINT "d %15" HOST_WIDEST_INT_PRINT "d\n",
 	   "Total", info.count, info.size);
   fprintf (stderr, "---------------------------------------------------------------------------------\n");
 }
diff -Naur gcc-4.9.2/gcc/cgraph.c gcc-4.9.2_curr/gcc/cgraph.c
--- ../gcc-4.9.2/gcc/cgraph.c	2014-06-20 05:57:53.000000000 -0400
+++ ./gcc/cgraph.c	2024-03-04 05:47:30.074589168 -0500
@@ -1976,7 +1976,7 @@
   fprintf (f, "  First run: %i\n", node->tp_first_run);
   fprintf (f, "  Function flags:");
   if (node->count)
-    fprintf (f, " executed "HOST_WIDEST_INT_PRINT_DEC"x",
+    fprintf (f, " executed " HOST_WIDEST_INT_PRINT_DEC "x",
 	     (HOST_WIDEST_INT)node->count);
   if (node->origin)
     fprintf (f, " nested in: %s", node->origin->asm_name ());
@@ -2028,7 +2028,7 @@
       fprintf (f, "%s/%i ", edge->caller->asm_name (),
 	       edge->caller->order);
       if (edge->count)
-	fprintf (f, "("HOST_WIDEST_INT_PRINT_DEC"x) ",
+	fprintf (f, "(" HOST_WIDEST_INT_PRINT_DEC "x) ",
 		 (HOST_WIDEST_INT)edge->count);
       if (edge->frequency)
 	fprintf (f, "(%.2f per call) ",
@@ -2055,7 +2055,7 @@
       if (edge->indirect_inlining_edge)
 	fprintf (f, "(indirect_inlining) ");
       if (edge->count)
-	fprintf (f, "("HOST_WIDEST_INT_PRINT_DEC"x) ",
+	fprintf (f, "(" HOST_WIDEST_INT_PRINT_DEC "x) ",
 		 (HOST_WIDEST_INT)edge->count);
       if (edge->frequency)
 	fprintf (f, "(%.2f per call) ",
diff -Naur gcc-4.9.2/gcc/config/elfos.h gcc-4.9.2_curr/gcc/config/elfos.h
--- ../gcc-4.9.2/gcc/config/elfos.h	2014-04-07 09:27:39.000000000 -0400
+++ ./gcc/config/elfos.h	2024-03-04 14:03:45.878152314 -0500
@@ -99,7 +99,7 @@
 
 #undef  ASM_OUTPUT_SKIP
 #define ASM_OUTPUT_SKIP(FILE, SIZE) \
-   fprintf ((FILE), "%s"HOST_WIDE_INT_PRINT_UNSIGNED"\n",\
+   fprintf ((FILE), "%s" HOST_WIDE_INT_PRINT_UNSIGNED "\n",\
 	    SKIP_ASM_OP, (SIZE))
 
 /* This is how to store into the string LABEL
@@ -167,7 +167,7 @@
     {									\
       fprintf ((FILE), "%s", COMMON_ASM_OP);				\
       assemble_name ((FILE), (NAME));					\
-      fprintf ((FILE), ","HOST_WIDE_INT_PRINT_UNSIGNED",%u\n",		\
+      fprintf ((FILE), "," HOST_WIDE_INT_PRINT_UNSIGNED ",%u\n",		\
 	       (SIZE), (ALIGN) / BITS_PER_UNIT);			\
     }									\
   while (0)
diff -Naur gcc-4.9.2/gcc/cp/cfns.h gcc-4.9.2_curr/gcc/cp/cfns.h
--- ../gcc-4.9.2/gcc/cp/cfns.h	2014-01-02 17:23:26.000000000 -0500
+++ ./gcc/cp/cfns.h	2024-03-04 15:43:09.182079704 -0500
@@ -89,7 +89,7 @@
       400, 400, 400, 400, 400, 400, 400, 400, 400, 400,
       400, 400, 400, 400, 400, 400, 400
     };
-  register int hval = len;
+  int hval = len;
 
   switch (hval)
     {
@@ -333,15 +333,15 @@
 
   if (len <= MAX_WORD_LENGTH && len >= MIN_WORD_LENGTH)
     {
-      register int key = hash (str, len);
+      int key = hash (str, len);
 
       if (key <= MAX_HASH_VALUE && key >= 0)
         {
-          register int index = lookup[key];
+          int index = lookup[key];
 
           if (index >= 0)
             {
-              register const char *s = wordlist[index];
+              const char *s = wordlist[index];
 
               if (*str == *s && !strcmp (str + 1, s + 1))
                 return s;
diff -Naur gcc-4.9.2/gcc/cppbuiltin.c gcc-4.9.2_curr/gcc/cppbuiltin.c
--- ../gcc-4.9.2/gcc/cppbuiltin.c	2014-01-24 18:13:49.000000000 -0500
+++ ./gcc/cppbuiltin.c	2024-03-04 05:48:52.897930752 -0500
@@ -129,7 +129,7 @@
 define_builtin_macros_for_type_sizes (cpp_reader *pfile)
 {
 #define define_type_sizeof(NAME, TYPE)                             \
-    cpp_define_formatted (pfile, NAME"="HOST_WIDE_INT_PRINT_DEC,   \
+    cpp_define_formatted (pfile, NAME "=" HOST_WIDE_INT_PRINT_DEC,   \
                           tree_to_uhwi (TYPE_SIZE_UNIT (TYPE)))
 
   define_type_sizeof ("__SIZEOF_INT__", integer_type_node);
diff -Naur gcc-4.9.2/gcc/dwarf2cfi.c gcc-4.9.2_curr/gcc/dwarf2cfi.c
--- ../gcc-4.9.2/gcc/dwarf2cfi.c	2014-01-02 17:23:26.000000000 -0500
+++ ./gcc/dwarf2cfi.c	2024-03-04 05:50:50.044609088 -0500
@@ -3176,7 +3176,7 @@
     case DW_CFA_offset_extended:
     case DW_CFA_offset_extended_sf:
       r = DWARF2_FRAME_REG_OUT (cfi->dw_cfi_oprnd1.dw_cfi_reg_num, 1);
-      fprintf (f, "\t.cfi_offset %lu, "HOST_WIDE_INT_PRINT_DEC"\n",
+      fprintf (f, "\t.cfi_offset %lu, " HOST_WIDE_INT_PRINT_DEC "\n",
 	       r, cfi->dw_cfi_oprnd2.dw_cfi_offset);
       break;
 
@@ -3199,7 +3199,7 @@
     case DW_CFA_def_cfa:
     case DW_CFA_def_cfa_sf:
       r = DWARF2_FRAME_REG_OUT (cfi->dw_cfi_oprnd1.dw_cfi_reg_num, 1);
-      fprintf (f, "\t.cfi_def_cfa %lu, "HOST_WIDE_INT_PRINT_DEC"\n",
+      fprintf (f, "\t.cfi_def_cfa %lu, " HOST_WIDE_INT_PRINT_DEC "\n",
 	       r, cfi->dw_cfi_oprnd2.dw_cfi_offset);
       break;
 
@@ -3234,13 +3234,13 @@
 	  fprintf (f, "\t.cfi_escape %#x,", DW_CFA_GNU_args_size);
 	  dw2_asm_output_data_uleb128_raw (cfi->dw_cfi_oprnd1.dw_cfi_offset);
 	  if (flag_debug_asm)
-	    fprintf (f, "\t%s args_size "HOST_WIDE_INT_PRINT_DEC,
+	    fprintf (f, "\t%s args_size " HOST_WIDE_INT_PRINT_DEC,
 		     ASM_COMMENT_START, cfi->dw_cfi_oprnd1.dw_cfi_offset);
 	  fputc ('\n', f);
 	}
       else
 	{
-	  fprintf (f, "\t.cfi_GNU_args_size "HOST_WIDE_INT_PRINT_DEC "\n",
+	  fprintf (f, "\t.cfi_GNU_args_size " HOST_WIDE_INT_PRINT_DEC "\n",
 		   cfi->dw_cfi_oprnd1.dw_cfi_offset);
 	}
       break;
diff -Naur gcc-4.9.2/gcc/dwarf2out.c gcc-4.9.2_curr/gcc/dwarf2out.c
--- ../gcc-4.9.2/gcc/dwarf2out.c	2014-10-01 16:57:44.000000000 -0400
+++ ./gcc/dwarf2out.c	2024-03-04 05:51:32.824613350 -0500
@@ -5324,8 +5324,8 @@
 	  fprintf (outfile, HOST_WIDE_INT_PRINT_UNSIGNED, AT_unsigned (a));
 	  break;
 	case dw_val_class_const_double:
-	  fprintf (outfile, "constant ("HOST_WIDE_INT_PRINT_DEC","\
-			    HOST_WIDE_INT_PRINT_UNSIGNED")",
+	  fprintf (outfile, "constant (" HOST_WIDE_INT_PRINT_DEC ","\
+			    HOST_WIDE_INT_PRINT_UNSIGNED ")",
 		   a->dw_attr_val.v.val_double.high,
 		   a->dw_attr_val.v.val_double.low);
 	  break;
diff -Naur gcc-4.9.2/gcc/gengtype-lex.c gcc-4.9.2_curr/gcc/gengtype-lex.c
--- ../gcc-4.9.2/gcc/gengtype-lex.c	2014-10-30 04:31:06.000000000 -0400
+++ ./gcc/gengtype-lex.c	2024-03-01 10:22:15.252241570 -0500
@@ -1406,9 +1406,9 @@
  */
 YY_DECL
 {
-	register yy_state_type yy_current_state;
-	register char *yy_cp, *yy_bp;
-	register int yy_act;
+	yy_state_type yy_current_state;
+	char *yy_cp, *yy_bp;
+	int yy_act;
     
 #line 65 "/d/gcc-4.9.3/gcc-4.9.3/gcc/gengtype-lex.l"
 
@@ -1466,7 +1466,7 @@
 yy_match:
 		do
 			{
-			register YY_CHAR yy_c = yy_ec[YY_SC_TO_UI(*yy_cp)];
+			YY_CHAR yy_c = yy_ec[YY_SC_TO_UI(*yy_cp)];
 			if ( yy_accept[yy_current_state] )
 				{
 				(yy_last_accepting_state) = yy_current_state;
@@ -2054,9 +2054,9 @@
  */
 static int yy_get_next_buffer (void)
 {
-    	register char *dest = YY_CURRENT_BUFFER_LVALUE->yy_ch_buf;
-	register char *source = (yytext_ptr);
-	register int number_to_move, i;
+    	char *dest = YY_CURRENT_BUFFER_LVALUE->yy_ch_buf;
+	char *source = (yytext_ptr);
+	int number_to_move, i;
 	int ret_val;
 
 	if ( (yy_c_buf_p) > &YY_CURRENT_BUFFER_LVALUE->yy_ch_buf[(yy_n_chars) + 1] )
@@ -2188,15 +2188,15 @@
 
     static yy_state_type yy_get_previous_state (void)
 {
-	register yy_state_type yy_current_state;
-	register char *yy_cp;
+	yy_state_type yy_current_state;
+	char *yy_cp;
     
 	yy_current_state = (yy_start);
 	yy_current_state += YY_AT_BOL();
 
 	for ( yy_cp = (yytext_ptr) + YY_MORE_ADJ; yy_cp < (yy_c_buf_p); ++yy_cp )
 		{
-		register YY_CHAR yy_c = (*yy_cp ? yy_ec[YY_SC_TO_UI(*yy_cp)] : 1);
+		YY_CHAR yy_c = (*yy_cp ? yy_ec[YY_SC_TO_UI(*yy_cp)] : 1);
 		if ( yy_accept[yy_current_state] )
 			{
 			(yy_last_accepting_state) = yy_current_state;
@@ -2221,10 +2221,10 @@
  */
     static yy_state_type yy_try_NUL_trans  (yy_state_type yy_current_state )
 {
-	register int yy_is_jam;
-    	register char *yy_cp = (yy_c_buf_p);
+	int yy_is_jam;
+    	char *yy_cp = (yy_c_buf_p);
 
-	register YY_CHAR yy_c = 1;
+	YY_CHAR yy_c = 1;
 	if ( yy_accept[yy_current_state] )
 		{
 		(yy_last_accepting_state) = yy_current_state;
@@ -2831,7 +2831,7 @@
 #ifndef yytext_ptr
 static void yy_flex_strncpy (char* s1, yyconst char * s2, int n )
 {
-	register int i;
+	int i;
 	for ( i = 0; i < n; ++i )
 		s1[i] = s2[i];
 }
@@ -2840,7 +2840,7 @@
 #ifdef YY_NEED_STRLEN
 static int yy_flex_strlen (yyconst char * s )
 {
-	register int n;
+	int n;
 	for ( n = 0; s[n]; ++n )
 		;
 
diff -Naur gcc-4.9.2/gcc/ipa-devirt.c gcc-4.9.2_curr/gcc/ipa-devirt.c
--- ../gcc-4.9.2/gcc/ipa-devirt.c	2014-10-04 22:50:01.000000000 -0400
+++ ./gcc/ipa-devirt.c	2024-03-04 07:11:43.885431243 -0500
@@ -1634,7 +1634,7 @@
     {
       fprintf (f, "    Contained in type:");
       print_generic_expr (f, ctx.outer_type, TDF_SLIM);
-      fprintf (f, " at offset "HOST_WIDE_INT_PRINT_DEC"\n",
+      fprintf (f, " at offset " HOST_WIDE_INT_PRINT_DEC "\n",
 	       ctx.offset);
     }
 
diff -Naur gcc-4.9.2/gcc/ipa-inline.c gcc-4.9.2_curr/gcc/ipa-inline.c
--- ../gcc-4.9.2/gcc/ipa-inline.c	2014-05-21 01:40:09.000000000 -0400
+++ ./gcc/ipa-inline.c	2024-03-04 07:12:08.282100340 -0500
@@ -1727,7 +1727,7 @@
 		   badness,
 		   edge->frequency / (double)CGRAPH_FREQ_BASE);
 	  if (edge->count)
-	    fprintf (dump_file," Called "HOST_WIDEST_INT_PRINT_DEC"x\n",
+	    fprintf (dump_file," Called " HOST_WIDEST_INT_PRINT_DEC "x\n",
 		     edge->count);
 	  if (dump_flags & TDF_DETAILS)
 	    edge_badness (edge, true);
diff -Naur gcc-4.9.2/gcc/ipa-profile.c gcc-4.9.2_curr/gcc/ipa-profile.c
--- ../gcc-4.9.2/gcc/ipa-profile.c	2014-01-02 17:23:26.000000000 -0500
+++ ./gcc/ipa-profile.c	2024-03-04 07:13:50.985443905 -0500
@@ -163,7 +163,7 @@
     {
       cumulated_time += histogram[i]->count * histogram[i]->time;
       cumulated_size += histogram[i]->size;
-      fprintf (file, "  "HOST_WIDEST_INT_PRINT_DEC": time:%i (%2.2f) size:%i (%2.2f)\n",
+      fprintf (file, "  " HOST_WIDEST_INT_PRINT_DEC ": time:%i (%2.2f) size:%i (%2.2f)\n",
 	       (HOST_WIDEST_INT) histogram[i]->count,
 	       histogram[i]->time,
 	       cumulated_time * 100.0 / overall_time,
@@ -516,7 +516,7 @@
 	{
 	  gcov_type min, cumulated_time = 0, cumulated_size = 0;
 
-	  fprintf (dump_file, "Overall time: "HOST_WIDEST_INT_PRINT_DEC"\n", 
+	  fprintf (dump_file, "Overall time: " HOST_WIDEST_INT_PRINT_DEC "\n", 
 		   (HOST_WIDEST_INT)overall_time);
 	  min = get_hot_bb_threshold ();
           for (i = 0; i < (int)histogram.length () && histogram[i]->count >= min;
@@ -525,7 +525,7 @@
 	      cumulated_time += histogram[i]->count * histogram[i]->time;
 	      cumulated_size += histogram[i]->size;
 	    }
-	  fprintf (dump_file, "GCOV min count: "HOST_WIDEST_INT_PRINT_DEC
+	  fprintf (dump_file, "GCOV min count: " HOST_WIDEST_INT_PRINT_DEC
 		   " Time:%3.2f%% Size:%3.2f%%\n", 
 		   (HOST_WIDEST_INT)min,
 		   cumulated_time * 100.0 / overall_time,
@@ -551,7 +551,7 @@
 	      cumulated_time += histogram[i]->count * histogram[i]->time;
 	      cumulated_size += histogram[i]->size;
 	    }
-	  fprintf (dump_file, "Determined min count: "HOST_WIDEST_INT_PRINT_DEC
+	  fprintf (dump_file, "Determined min count: " HOST_WIDEST_INT_PRINT_DEC
 		   " Time:%3.2f%% Size:%3.2f%%\n", 
 		   (HOST_WIDEST_INT)threshold,
 		   cumulated_time * 100.0 / overall_time,
diff -Naur gcc-4.9.2/gcc/ipa-prop.c gcc-4.9.2_curr/gcc/ipa-prop.c
--- ../gcc-4.9.2/gcc/ipa-prop.c	2014-06-26 10:33:06.000000000 -0400
+++ ./gcc/ipa-prop.c	2024-03-04 07:14:46.688782787 -0500
@@ -244,7 +244,7 @@
 	{
 	  fprintf (f, "KNOWN TYPE: base  ");
 	  print_generic_expr (f, jump_func->value.known_type.base_type, 0);
-	  fprintf (f, ", offset "HOST_WIDE_INT_PRINT_DEC", component ",
+	  fprintf (f, ", offset " HOST_WIDE_INT_PRINT_DEC ", component ",
 		   jump_func->value.known_type.offset);
 	  print_generic_expr (f, jump_func->value.known_type.component_type, 0);
 	  fprintf (f, "\n");
@@ -284,7 +284,7 @@
       else if (type == IPA_JF_ANCESTOR)
 	{
 	  fprintf (f, "ANCESTOR: ");
-	  fprintf (f, "%d, offset "HOST_WIDE_INT_PRINT_DEC", ",
+	  fprintf (f, "%d, offset " HOST_WIDE_INT_PRINT_DEC ", ",
 		   jump_func->value.ancestor.formal_id,
 		   jump_func->value.ancestor.offset);
 	  print_generic_expr (f, jump_func->value.ancestor.type, 0);
diff -Naur gcc-4.9.2/gcc/loop-iv.c gcc-4.9.2_curr/gcc/loop-iv.c
--- ../gcc-4.9.2/gcc/loop-iv.c	2014-01-17 05:47:59.000000000 -0500
+++ ./gcc/loop-iv.c	2024-03-04 07:15:59.312123355 -0500
@@ -2297,7 +2297,7 @@
   if (andmax)
     nmax = MIN (nmax, andmax);
   if (dump_file)
-    fprintf (dump_file, ";; Determined upper bound "HOST_WIDEST_INT_PRINT_DEC".\n",
+    fprintf (dump_file, ";; Determined upper bound " HOST_WIDEST_INT_PRINT_DEC ".\n",
 	     nmax);
   return nmax;
 }
diff -Naur gcc-4.9.2/gcc/profile.c gcc-4.9.2_curr/gcc/profile.c
--- ../gcc-4.9.2/gcc/profile.c	2014-02-01 03:40:31.000000000 -0500
+++ ./gcc/profile.c	2024-03-04 07:17:34.928799547 -0500
@@ -318,7 +318,7 @@
 	      if (dump_file)
 		{
 		  fprintf (dump_file,
-		  	   "Edge %i->%i is inconsistent, count"HOST_WIDEST_INT_PRINT_DEC,
+		  	   "Edge %i->%i is inconsistent, count" HOST_WIDEST_INT_PRINT_DEC,
 			   e->src->index, e->dest->index, e->count);
 		  dump_bb (dump_file, e->src, 0, TDF_DETAILS);
 		  dump_bb (dump_file, e->dest, 0, TDF_DETAILS);
diff -Naur gcc-4.9.2/gcc/stor-layout.c gcc-4.9.2_curr/gcc/stor-layout.c
--- ../gcc-4.9.2/gcc/stor-layout.c	2014-10-15 06:27:26.000000000 -0400
+++ ./gcc/stor-layout.c	2024-03-04 12:29:32.380922449 -0500
@@ -236,7 +236,7 @@
   fntype = build_function_type (return_type, param_type_list);
 
   /* Build the function declaration.  */
-  sprintf (buf, "SZ"HOST_WIDE_INT_PRINT_UNSIGNED, fnno++);
+  sprintf (buf, "SZ" HOST_WIDE_INT_PRINT_UNSIGNED, fnno++);
   fnname = get_file_function_name (buf);
   fndecl = build_decl (input_location, FUNCTION_DECL, fnname, fntype);
   for (t = param_decl_list; t; t = DECL_CHAIN (t))
diff -Naur gcc-4.9.2/gcc/toplev.c gcc-4.9.2_curr/gcc/toplev.c
--- ../gcc-4.9.2/gcc/toplev.c	2014-10-16 09:49:11.000000000 -0400
+++ ./gcc/toplev.c	2024-03-04 12:32:43.340941473 -0500
@@ -1041,7 +1041,7 @@
 	}
 
       fprintf (stack_usage_file,
-	       "%s:%d:%d:%s\t"HOST_WIDE_INT_PRINT_DEC"\t%s\n",
+	       "%s:%d:%d:%s\t" HOST_WIDE_INT_PRINT_DEC "\t%s\n",
 	       lbasename (loc.file),
 	       loc.line,
 	       loc.column,
diff -Naur gcc-4.9.2/gcc/tree-browser.c gcc-4.9.2_curr/gcc/tree-browser.c
--- ../gcc-4.9.2/gcc/tree-browser.c	2014-01-02 17:23:26.000000000 -0500
+++ ./gcc/tree-browser.c	2024-03-04 15:26:34.698647302 -0500
@@ -938,7 +938,7 @@
 
       while (--copy > 0)
         {
-          register int c = getc (stream);
+          int c = getc (stream);
           if (c == EOF)
             goto lose;
           else if ((*p++ = c) == '\n')
diff -Naur gcc-4.9.2/gcc/tree-ssa-reassoc.c gcc-4.9.2_curr/gcc/tree-ssa-reassoc.c
--- ../gcc-4.9.2/gcc/tree-ssa-reassoc.c	2014-10-17 06:53:36.000000000 -0400
+++ ./gcc/tree-ssa-reassoc.c	2024-03-04 13:44:24.648036633 -0500
@@ -4211,7 +4211,7 @@
 		      if (elt < vec_len - 1)
 			fputs (" * ", dump_file);
 		    }
-		  fprintf (dump_file, ")^"HOST_WIDE_INT_PRINT_DEC"\n",
+		  fprintf (dump_file, ")^" HOST_WIDE_INT_PRINT_DEC "\n",
 			   power);
 		}
 	    }
@@ -4245,7 +4245,7 @@
 		  if (elt < vec_len - 1)
 		    fputs (" * ", dump_file);
 		}
-	      fprintf (dump_file, ")^"HOST_WIDE_INT_PRINT_DEC"\n", power);
+	      fprintf (dump_file, ")^" HOST_WIDE_INT_PRINT_DEC "\n", power);
 	    }
 
 	  reassociate_stats.pows_created++;
diff -Naur gcc-4.9.2/gcc/value-prof.c gcc-4.9.2_curr/gcc/value-prof.c
--- ../gcc-4.9.2/gcc/value-prof.c	2014-01-02 17:23:26.000000000 -0500
+++ ./gcc/value-prof.c	2024-03-04 13:48:09.721392388 -0500
@@ -266,10 +266,10 @@
 	   unsigned int i;
 	   fprintf (dump_file, " [");
            for (i = 0; i < hist->hdata.intvl.steps; i++)
-	     fprintf (dump_file, " %d:"HOST_WIDEST_INT_PRINT_DEC,
+	     fprintf (dump_file, " %d:" HOST_WIDEST_INT_PRINT_DEC,
 		      hist->hdata.intvl.int_start + i,
 		      (HOST_WIDEST_INT) hist->hvalue.counters[i]);
-	   fprintf (dump_file, " ] outside range:"HOST_WIDEST_INT_PRINT_DEC,
+	   fprintf (dump_file, " ] outside range:" HOST_WIDEST_INT_PRINT_DEC,
 		    (HOST_WIDEST_INT) hist->hvalue.counters[i]);
 	}
       fprintf (dump_file, ".\n");
@@ -279,8 +279,8 @@
       fprintf (dump_file, "Pow2 counter ");
       if (hist->hvalue.counters)
 	{
-	   fprintf (dump_file, "pow2:"HOST_WIDEST_INT_PRINT_DEC
-		    " nonpow2:"HOST_WIDEST_INT_PRINT_DEC,
+	   fprintf (dump_file, "pow2:" HOST_WIDEST_INT_PRINT_DEC
+		    " nonpow2:" HOST_WIDEST_INT_PRINT_DEC,
 		    (HOST_WIDEST_INT) hist->hvalue.counters[0],
 		    (HOST_WIDEST_INT) hist->hvalue.counters[1]);
 	}
@@ -291,9 +291,9 @@
       fprintf (dump_file, "Single value ");
       if (hist->hvalue.counters)
 	{
-	   fprintf (dump_file, "value:"HOST_WIDEST_INT_PRINT_DEC
-		    " match:"HOST_WIDEST_INT_PRINT_DEC
-		    " wrong:"HOST_WIDEST_INT_PRINT_DEC,
+	   fprintf (dump_file, "value:" HOST_WIDEST_INT_PRINT_DEC
+		    " match:" HOST_WIDEST_INT_PRINT_DEC
+		    " wrong:" HOST_WIDEST_INT_PRINT_DEC,
 		    (HOST_WIDEST_INT) hist->hvalue.counters[0],
 		    (HOST_WIDEST_INT) hist->hvalue.counters[1],
 		    (HOST_WIDEST_INT) hist->hvalue.counters[2]);
@@ -305,8 +305,8 @@
       fprintf (dump_file, "Average value ");
       if (hist->hvalue.counters)
 	{
-	   fprintf (dump_file, "sum:"HOST_WIDEST_INT_PRINT_DEC
-		    " times:"HOST_WIDEST_INT_PRINT_DEC,
+	   fprintf (dump_file, "sum:" HOST_WIDEST_INT_PRINT_DEC
+		    " times:" HOST_WIDEST_INT_PRINT_DEC,
 		    (HOST_WIDEST_INT) hist->hvalue.counters[0],
 		    (HOST_WIDEST_INT) hist->hvalue.counters[1]);
 	}
@@ -317,7 +317,7 @@
       fprintf (dump_file, "IOR value ");
       if (hist->hvalue.counters)
 	{
-	   fprintf (dump_file, "ior:"HOST_WIDEST_INT_PRINT_DEC,
+	   fprintf (dump_file, "ior:" HOST_WIDEST_INT_PRINT_DEC,
 		    (HOST_WIDEST_INT) hist->hvalue.counters[0]);
 	}
       fprintf (dump_file, ".\n");
@@ -327,9 +327,9 @@
       fprintf (dump_file, "Constant delta ");
       if (hist->hvalue.counters)
 	{
-	   fprintf (dump_file, "value:"HOST_WIDEST_INT_PRINT_DEC
-		    " match:"HOST_WIDEST_INT_PRINT_DEC
-		    " wrong:"HOST_WIDEST_INT_PRINT_DEC,
+	   fprintf (dump_file, "value:" HOST_WIDEST_INT_PRINT_DEC
+		    " match:" HOST_WIDEST_INT_PRINT_DEC
+		    " wrong:" HOST_WIDEST_INT_PRINT_DEC,
 		    (HOST_WIDEST_INT) hist->hvalue.counters[0],
 		    (HOST_WIDEST_INT) hist->hvalue.counters[1],
 		    (HOST_WIDEST_INT) hist->hvalue.counters[2]);
@@ -340,9 +340,9 @@
       fprintf (dump_file, "Indirect call ");
       if (hist->hvalue.counters)
 	{
-	   fprintf (dump_file, "value:"HOST_WIDEST_INT_PRINT_DEC
-		    " match:"HOST_WIDEST_INT_PRINT_DEC
-		    " all:"HOST_WIDEST_INT_PRINT_DEC,
+	   fprintf (dump_file, "value:" HOST_WIDEST_INT_PRINT_DEC
+		    " match:" HOST_WIDEST_INT_PRINT_DEC
+		    " all:" HOST_WIDEST_INT_PRINT_DEC,
 		    (HOST_WIDEST_INT) hist->hvalue.counters[0],
 		    (HOST_WIDEST_INT) hist->hvalue.counters[1],
 		    (HOST_WIDEST_INT) hist->hvalue.counters[2]);
@@ -353,7 +353,7 @@
       fprintf (dump_file, "Time profile ");
       if (hist->hvalue.counters)
       {
-        fprintf (dump_file, "time:"HOST_WIDEST_INT_PRINT_DEC,
+        fprintf (dump_file, "time:" HOST_WIDEST_INT_PRINT_DEC,
                  (HOST_WIDEST_INT) hist->hvalue.counters[0]);
       }
       fprintf (dump_file, ".\n");
@@ -1533,8 +1533,8 @@
       print_generic_expr (dump_file, direct_call->decl, TDF_SLIM);
       fprintf (dump_file, " transformation on insn postponned to ipa-profile");
       print_gimple_stmt (dump_file, stmt, 0, TDF_SLIM);
-      fprintf (dump_file, "hist->count "HOST_WIDEST_INT_PRINT_DEC
-	       " hist->all "HOST_WIDEST_INT_PRINT_DEC"\n", count, all);
+      fprintf (dump_file, "hist->count " HOST_WIDEST_INT_PRINT_DEC
+	       " hist->all " HOST_WIDEST_INT_PRINT_DEC "\n", count, all);
     }
 
   return true;
