diff -Naur gcc-4.9.2/gcc/output.h gcc-4.9.2_bak/gcc/output.h
--- gcc-4.9.2/gcc/output.h	2014-01-20 15:13:10.000000000 -0500
+++ gcc-4.9.2_bak/gcc/output.h	2024-04-01 04:58:02.053396841 -0400
@@ -117,7 +117,7 @@
 
 /* Output a string of assembler code, substituting numbers, strings
    and fixed syntactic prefixes.  */
-#if GCC_VERSION >= 3004
+#if GCC_VERSION >= 3004 && 0
 #define ATTRIBUTE_ASM_FPRINTF(m, n) __attribute__ ((__format__ (__asm_fprintf__, m, n))) ATTRIBUTE_NONNULL(m)
 #else
 #define ATTRIBUTE_ASM_FPRINTF(m, n) ATTRIBUTE_NONNULL(m)
diff -Naur gcc-4.9.2/gcc/reload1.c gcc-4.9.2_bak/gcc/reload1.c
--- gcc-4.9.2/gcc/reload1.c	2014-02-26 16:57:40.000000000 -0500
+++ gcc-4.9.2_bak/gcc/reload1.c	2024-04-01 04:58:02.063396842 -0400
@@ -441,7 +441,7 @@
 
   while (memory_address_p (QImode, tem))
     {
-      spill_indirect_levels++;
+      spill_indirect_levels += 1;
       tem = gen_rtx_MEM (Pmode, tem);
     }
 
