--- ./usr/include/android/versioning.h.orig	2026-02-26 23:44:31.515531551 -0500
+++ ./usr/include/android/versioning.h	2026-02-21 00:49:07.418716448 -0500
@@ -31,7 +31,13 @@
 // minSdkVersion is less than the __what argument. This allows the libraries to
 // load even on systems too old to contain the API, but calls must be guarded
 // with `__builtin_available(android api_level, *)` to avoid segfaults.
-#define __BIONIC_AVAILABILITY(__what, ...) __attribute__((__availability__(android,__what __VA_OPT__(,) __VA_ARGS__)))
+#if __has_attribute(availability)
+#define __BIONIC_AVAILABILITY(__what) __attribute__((__availability__(android,__what)))
+#define __BIONIC_AVAILABILITY_MSG(__what, __msg) __attribute__((__availability__(android,__what,__msg)))
+#else
+#define __BIONIC_AVAILABILITY(__what)
+#define __BIONIC_AVAILABILITY_MSG(__what, __msg)
+#endif
 
 // When the caller is using weak API references, we should expose the decls for
 // APIs which are not available in the caller's minSdkVersion, otherwise there's
@@ -44,7 +50,13 @@
 // `__builtin_available()` for run-time availability checking, and so would not
 // compile in this mode (or worse, if the build doesn't use
 // -Werror=unguarded-availability, it would build but crash at runtime).
-#define __BIONIC_AVAILABILITY(__what, ...) __attribute__((__availability__(android,strict,__what __VA_OPT__(,) __VA_ARGS__)))
+#if __has_attribute(availability)
+#define __BIONIC_AVAILABILITY(__what) __attribute__((__availability__(android,strict,__what)))
+#define __BIONIC_AVAILABILITY_MSG(__what, __msg) __attribute__((__availability__(android,strict,__what,__msg)))
+#else
+#define __BIONIC_AVAILABILITY(__what)
+#define __BIONIC_AVAILABILITY_MSG(__what, __msg)
+#endif
 
 // When the caller is using strict API references, we hide APIs which are not
 // available in the caller's minSdkVersion. This is a bionic-only deviation in
@@ -64,8 +76,8 @@
 #define __INTRODUCED_IN(api_level) __BIONIC_AVAILABILITY(introduced=api_level)
 #pragma clang diagnostic pop
 
-#define __DEPRECATED_IN(api_level, msg) __BIONIC_AVAILABILITY(deprecated=api_level, message=msg)
-#define __REMOVED_IN(api_level, msg) __BIONIC_AVAILABILITY(obsoleted=api_level, message=msg)
+#define __DEPRECATED_IN(api_level, msg) __BIONIC_AVAILABILITY_MSG(deprecated=api_level, message=msg)
+#define __REMOVED_IN(api_level, msg) __BIONIC_AVAILABILITY_MSG(obsoleted=api_level, message=msg)
 
 // The same availability attribute can't be annotated multiple times. Therefore, the macros are
 // defined for the configuration that it is valid for so that declarations like the below doesn't
