--- a/syntax/default.nanorc
+++ b/syntax/default.nanorc
@@ -24,10 +24,16 @@
 color brightgreen "^\[[^][]+\]$"
 
 # Comments.
-color cyan "^[[:blank:]]*#.*"
+color cyan "(^|[[:blank:]])#.*"
 
+# Strings.
+color brightyellow ""([^"\]|\\.)*"|'([^'\]|\\.)*'"
+
 # Make hard (non-breaking) spaces noticeable.
 color ,#444 " "
 
 # Control codes.
 color orange "[[:cntrl:]]"
+
+# Trailing whitespace.
+color ,green "[[:space:]]+$"
--- a/syntax/makefile.nanorc
+++ b/syntax/makefile.nanorc
@@ -1,6 +1,6 @@
 ## Syntax highlighting for Makefiles.
 
-syntax makefile "(^|/)((GNU)?m|M)akefile[^/]*$|\.(make|mk)$"
+syntax makefile "(^|/)((GNU)?m|M)akefile[^/]*$|\.(make|mak|mk)$"
 magic "makefile script"
 
 tabgives "	"
--- a/syntax/nanorc.nanorc
+++ b/syntax/nanorc.nanorc
@@ -7,7 +7,7 @@
 color brightred ".*"
 
 # Color names
-color yellow "^[[:blank:]]*(i?color|set[[:blank:]]+(error|function|key|mini|number|prompt|scroller|selected|spotlight|status|stripe|title)color)[[:blank:]]+(bold,)?(italic,)?((bright|light)?(white|black|red|blue|green|yellow|magenta|cyan)|normal|pink|purple|mauve|lagoon|mint|lime|peach|orange|latte|rosy|beet|plum|sea|sky|slate|teal|sage|brown|ocher|sand|tawny|brick|crimson|grey|gray|#[[:xdigit:]]{3})?(,((light)?(white|black|red|blue|green|yellow|magenta|cyan)|normal|pink|purple|mauve|lagoon|mint|lime|peach|orange|latte|rosy|beet|plum|sea|sky|slate|teal|sage|brown|ocher|sand|tawny|brick|crimson|grey|gray|#[[:xdigit:]]{3}))?\>"
+color yellow "^[[:blank:]]*(i?color|set[[:blank:]]+(error|function|key|mini|number|prompt|scroller|selected|spotlight|status|stripe|title)color)[[:blank:]]+(bold,)?(italic,)?((bright|light)?(white|black|red|blue|green|yellow|magenta|cyan)|normal|pink|purple|mauve|lagoon|mint|lime|peach|orange|latte|rosy|beet|plum|sea|sky|slate|teal|sage|brown|ocher|sand|tawny|brick|crimson|grey|gray|#[[:xdigit:]]{3})?(,((bright|light)?(white|black|red|blue|green|yellow|magenta|cyan)|normal|pink|purple|mauve|lagoon|mint|lime|peach|orange|latte|rosy|beet|plum|sea|sky|slate|teal|sage|brown|ocher|sand|tawny|brick|crimson|grey|gray|#[[:xdigit:]]{3}))?\>"
 
 # The arguments of two special commands
 color bold,purple "^[[:blank:]]*include[[:blank:]]+[^[:blank:]"]+"
--- a/syntax/sh.nanorc
+++ b/syntax/sh.nanorc
@@ -20,7 +20,7 @@
 color green "-(eq|ne|gt|lt|ge|le|ef|ot|nt)\>"
 
 # Short and long options.
-color brightmagenta "[[:blank:]](-[[:alpha:]]|--\<[[:alpha:]-]+)\>"
+color brightmagenta "[[:blank:]](-[A-Za-z0-9]|--\<[A-Za-z0-9-]+)\>"
 
 # Common commands.
 color brightblue "\<(awk|cat|cd|ch(grp|mod|own)|cp|cut|echo|env|grep|head|install|ln|make|mkdir|mv|popd|printf|pushd|rm|rmdir|sed|set|sort|tail|tar|touch|umask|unset)\>"
@@ -31,11 +31,11 @@
 # More complicated variable names; handles braces and replacements and arrays.
 color brightred "\$\{[#!]?([-@*#?$!]|[[:digit:]]+|[[:alpha:]_][[:alnum:]_]*)(\[([[:blank:]]*[[:alnum:]_]+[[:blank:]]*|@)\])?(([#%/]|:?[-=?+])[^}]*\}|\[|\})"
 
+# Strings.
+color brightyellow ""([^"\]|\\.)*"|'([^'\]|\\.)*'"
+
 # Comments.
 color cyan "(^|[[:blank:]])#.*"
 
-# Strings.
-color brightyellow ""([^"\]|\\.)*"|'([^'\]|\\.)*'"
-
 # Trailing whitespace.
 color ,green "[[:space:]]+$"
--- a/syntax/patch.nanorc
+++ b/syntax/patch.nanorc
@@ -3,7 +3,7 @@
 ## Original author:  Mike Frysinger
 ## License:  GPL version 3 or newer
 
-syntax patch "\.(patch|diff|debdiff|rej)$"
+syntax patch "\.(patch|diff|debdiff|rej|disabled)$"
 magic "diff output"
 # There is no official support for comments in patch files.
 comment ""
