--- ./usr/include/sys/cdefs.h.orig
+++ ./usr/include/sys/cdefs.h
@@ -353,12 +353,20 @@
 /* Used to rename functions so that the compiler emits a call to 'x' rather than the function this was applied to. */
 #define __RENAME(x) __asm__(#x)
 
-#if __has_builtin(__builtin_umul_overflow) || __GNUC__ >= 5
-#if defined(__LP64__)
-#define __size_mul_overflow(a, b, result) __builtin_umull_overflow(a, b, result)
-#else
-#define __size_mul_overflow(a, b, result) __builtin_umul_overflow(a, b, result)
-#endif
+#if defined __has_builtin
+# if __has_builtin(__builtin_umul_overflow)
+#  if defined(__LP64__)
+#   define __size_mul_overflow(a, b, result) __builtin_umull_overflow(a, b, result)
+#  else
+#   define __size_mul_overflow(a, b, result) __builtin_umul_overflow(a, b, result)
+#  endif
+# endif
+#elif __GNUC__ >= 5
+# if defined(__LP64__)
+#  define __size_mul_overflow(a, b, result) __builtin_umull_overflow(a, b, result)
+# else
+#  define __size_mul_overflow(a, b, result) __builtin_umul_overflow(a, b, result)
+# endif
 #else
 extern __inline__ __always_inline __attribute__((gnu_inline))
 int __size_mul_overflow(__SIZE_TYPE__ a, __SIZE_TYPE__ b, __SIZE_TYPE__ *result) {
